dask/tests/test_core.py

```diff
@@ -45,62 +42,114 @@ def test_preorder_traversal():
     assert list(preorder_traversal(t)) == [add, sum, list, 1, 2, 3]
 
 
-def test_get():
-    assert get(d, ':x') == 1
-    assert get(d, ':y') == 2
-    assert get(d, ':z') == 3
-    assert get(d, 'pass-through') == 'pass-through'
+class TestGet(unittest.TestCase):
+    """TestGet aims to enforce all known expectations of get functions. It can
+    be imported and used to test a foreign `get` function by setting the class
+    variable `get_function`, as shown below:
 
+    >>> import unittest  # doctest: +SKIP
+    >>> from dask.tests.test_core import TestGet  # doctest: +SKIP
+    >>> TestGet.get_function = my_get_function  # doctest: +SKIP
+    >>> loader = unittest.TestLoader()  # doctest: +SKIP
+    >>> suite = loader.loadTestsFromTestCase(TestGet)  # doctest: +SKIP
+    >>> unittest.TextTestRunner().run(suite)  # doctest: +SKIP
+    """
 
-def test_memoized_get():
-    try:
-        import toolz
-    except ImportError:
-        return
-    cache = dict()
-    getm = toolz.memoize(get, cache=cache, key=lambda args, kwargs: args[1:])
+    @classmethod
+    def setUpClass(cls):
+        if not hasattr(cls, 'get_function'):
+            cls.get_function = core.get
+
+    @staticmethod
+    def get(d, key, *args, **kwargs):
+        PY2 = sys.version_info[0] == 2
+        if PY2:
+            return TestGet.get_function.im_func(d, key, *args, **kwargs)
+        else:
+            return TestGet.get_function(d, key, *args, **kwargs)
```

Comment:
These still seem fairly complex to me.  Explicitly setting the following seems like it would work fine:

```python
class TestGet(unittest.TestCase):
    get = core.get
    ...
```

```python
class CustomTestGet(TestGet):
    get = myget
```